ssm_jax/utils.py

```diff
@@ -40,3 +44,128 @@ def pad(seq, len):
 
 def monotonically_increasing(x, atol=0):
     return jnp.all(jnp.diff(x) > -atol)
+
+
+def find_closest_cluster(X, centroids):
+
+    def assign(x):
+        distances = vmap(jnp.linalg.norm)(centroids - x)
+        min_distance = jnp.min(distances)
+        label = jnp.argmin(distances)
+        return label, min_distance
+
+    return vmap(assign)(X)
+
+
+def kmeans_plusplus_initialization(key, X, num_clusters, num_local_trials=None):
```

Comment:
Separate out kmeans stuff, as we discussed

Reply:
Removed them for now